authorization/resource/service/resource_service_blackbox_test.go

```diff
@@ -102,6 +104,131 @@ func (s *resourceServiceBlackBoxTest) TestRegisterReadDeleteResourceWithParentOK
 }
 
+func (s *resourceServiceBlackBoxTest) TestDeleteResourceOK() {
+
+	// Create test data
+
+	g := s.DBTestSuite.NewTestGraph()
+	org := g.CreateOrganization()
+
+	spaceToDelete := g.CreateSpace(org)
+	spaceToDelete.AddAdmin(g.CreateUser(g.ID("admin-user")))
+
+	teamToDeleteA := g.CreateTeam(spaceToDelete)
+	teamToDeleteB := g.CreateTeam(spaceToDelete)
+
+	childResourceToDelete := g.CreateResource(spaceToDelete)
+	grandchildResourceToDelete := g.CreateResource(childResourceToDelete)
+
+	spaceRoleMappingToDelete := g.CreateRoleMapping(spaceToDelete)
+	grandchildRoleMappingToDelete := g.CreateRoleMapping(spaceToDelete)
+
+	spaceToStay := g.CreateSpace(org)
+	teamToStay := g.CreateTeam(spaceToStay)
+	spaceRoleMappingToStay := g.CreateRoleMapping(spaceToStay)
+
+	// Check test data
+
+	s.checkIdentity(true, teamToDeleteA.TeamID().String())
+	s.checkIdentity(true, teamToDeleteB.TeamID().String())
+
+	s.checkResource(true, spaceToDelete.SpaceID())
+	s.checkResource(true, teamToDeleteA.ResourceID())
+	s.checkResource(true, teamToDeleteB.ResourceID())
+	s.checkResource(true, childResourceToDelete.ResourceID())
+	s.checkResource(true, grandchildResourceToDelete.ResourceID())
+
+	s.checkIdentityRole(2, spaceToDelete.SpaceID())
+	s.checkIdentityRole(2, teamToDeleteA.ResourceID())
+	s.checkIdentityRole(2, teamToDeleteB.ResourceID())
+
+	s.checkRoleMapping(true, spaceRoleMappingToDelete.RoleMapping().RoleMappingID)
+	s.checkRoleMapping(true, grandchildRoleMappingToDelete.RoleMapping().RoleMappingID)
+
+	// Delete the space
+
+	err := s.resourceService.Delete(s.Ctx, spaceToDelete.SpaceID())
+	require.NoError(s.T(), err)
+
+	// Check all related artifacts for the space are gone
+
+	s.checkIdentity(false, teamToDeleteA.TeamID().String())
+	s.checkIdentity(false, teamToDeleteB.TeamID().String())
+
+	s.checkResource(false, spaceToDelete.SpaceID())
+	s.checkResource(false, teamToDeleteA.ResourceID())
+	s.checkResource(false, teamToDeleteB.ResourceID())
+
+	s.checkResource(false, childResourceToDelete.ResourceID())
+	s.checkResource(false, grandchildResourceToDelete.ResourceID())
+
+	s.checkIdentityRole(0, spaceToDelete.SpaceID())
+	s.checkIdentityRole(0, teamToDeleteA.ResourceID())
+	s.checkIdentityRole(0, teamToDeleteB.ResourceID())
+
+	s.checkRoleMapping(false, spaceRoleMappingToDelete.RoleMapping().RoleMappingID)
+	s.checkRoleMapping(false, grandchildRoleMappingToDelete.RoleMapping().RoleMappingID)
+
+	// Check all not-related artifacts are still present
+
+	s.checkResource(true, spaceToStay.SpaceID())
+	s.checkResource(true, teamToStay.ResourceID())
+
+	s.checkIdentity(true, g.UserByID("admin-user").Identity().ID.String())
+	s.checkIdentity(true, teamToStay.TeamID().String())
+
+	s.checkIdentityRole(1, spaceToStay.SpaceID())
+	s.checkIdentityRole(1, teamToStay.ResourceID())
+
+	s.checkRoleMapping(true, spaceRoleMappingToStay.RoleMapping().RoleMappingID)
+}
+
+func (s *resourceServiceBlackBoxTest) TestDeleteResourceWithCycleReferencesFails() {
```

Comment:
Yeah... I agree that we need to check cycle references when creating or updating a resource. Let's create an issue for that. But I would still keep the check here too just as an addition safety net. Just because if there is a cycle reference then this method will run in infinite loop.